test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/TagHelperDescriptorFactoryTest.cs

```diff
@@ -1755,6 +1779,14 @@ private class MultipleValidHtmlAttributePrefix : TagHelper
 
             [HtmlAttributeName("valid-name6")]
             public string StringProperty { get; set; }
+
+            public IDictionary<string, int> GetOnlyDictionaryProperty { get; }
+
+            [HtmlAttributeName("valid-name8")]
+            public IDictionary<string, object> GetOnlyDictionaryPropertyWithAttributeName { get; }
+
+            [HtmlAttributeName("ignored-name9", DictionaryAttributePrefix = "valid-prefix9")]
```

Comment:
This is a realllly weird case. I can see users providing `ignored-name9` and `valid-prefix9` and trying to use `ignored-name9` and having no clue why it's not binding. Suggest modifying `HtmlAttributeName` to provide a `public string Name { get; set; }` and a 0 and 1 ctor overload option. That then moves the `isNullOrEmpty` check in the 1 arg `name` ctor over to the `TagHelperDescriptorFactory` or a helper method on `HtmlAttributeName`; whatever works. 

Nice part of this is it'll let us get rid of the ugliness of `string.IsNullOrEmpty` in the `HtmlAttributeName` ctor. We were doing a bad practice since it's an attribute :cry: